### PR TITLE
feat: Add TypeScript Bindings and Improve Documentation

### PR DESCRIPTION
**Pull request by cuervolu**
## Changes in this PR

This pull request introduces several significant improvements to the OAuth plugin:

1. **TypeScript Bindings**: 
   - Created TypeScript bindings for the OAuth plugin, enabling easier integration in TypeScript-based Tauri applications.
   - Implemented functions like `start()`, `cancel()`, `onUrl()`, and `onInvalidUrl()` for TypeScript users.

2. **Updated Vanilla Example**:
   - Added a new example using the Tauri V2 vanilla template.
   - Demonstrates usage of the OAuth plugin in both Rust and TypeScript contexts.
   - Includes UI elements to start OAuth servers and display results.

3. **Improved README**:
   - Enhanced project description and rationale.
   - Added clear installation instructions for both Rust and npm/yarn.
   - Provided usage examples in Rust and TypeScript.
   - Included a configuration section with code samples.
   - Added important security considerations.

## Notes for Reviewers
- Please pay special attention to the TypeScript bindings to ensure they correctly mirror the Rust functionality.
- The vanilla example might benefit from additional error handling or edge case considerations.
- We need to verify that all code examples in the README work as expected. Please test these examples to ensure they are accurate and functional.
- An important next step is to publish the npm package for the TypeScript bindings. As i'm not familiar with this process, guidance or assistance with publishing to npm would be greatly appreciated.
- Any suggestions for further improving the documentation or examples are welcome!

### CODE REVIEW

**Comment by FabianLars** [comment]
sorry for the delay here, i'll try to get back to you as soon as i can

File: README.md [hunk 1]
```diff
@@ -1,8 +1,137 @@
...
-Needed because many sites such as Google and GitHub don't allow custom URI schemes ("deep link") as redirect URLs.
+A minimalistic Rust library and Tauri plugin for handling browser-based OAuth flows in desktop
+applications. This plugin spawns a temporary localhost server to capture OAuth redirects, solving
+the challenge of using OAuth with desktop apps.
 
-See https://github.com/FabianLars/tauri-plugin-deep-link for an alternative based on deep linking. This one will automatically start your app if there is no open instance.
+## Why This Plugin?
+
+Many OAuth providers (like Google and GitHub) don't allow custom URI schemes ("deep links") as
+redirect URLs for security reasons. This plugin provides a solution by:
```
Comment (FabianLars) [inline README.md]:
```suggestion
redirect URLs. This plugin provides a solution by:
```
tbh feels more like most of them don't support it just because. I'd prefer not to say it's for security reasons so that i'm not asked to explain what those are 😅

File: README.md [hunk 2]
```diff
@@ -1,8 +1,137 @@
...
+> **Note**: For an alternative approach using deep linking,
+> see [tauri-plugin-deep-link](https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/deep-link). The deep-link
+> plugin can automatically start your app if there's no open instance.
+
+## Installation
+
+```toml
+# Cargo.toml
+[dependencies]
+tauri-plugin-oauth = "0.0.0-alpha.0"
```
Comment (FabianLars) [inline README.md]:
```suggestion
tauri-plugin-oauth = "2.0.0-rc.0"
```

File: README.md [hunk 3]
```diff
@@ -1,8 +1,137 @@
...
+```
+
+For Tauri projects using npm or yarn:
+
+```bash
+npm install @fabianlars/tauri-plugin-oauth
+# or
+yarn add @fabianlars/tauri-plugin-oauth
```
Comment (FabianLars) [inline README.md]:
```suggestion
npm install @fabianlars/tauri-plugin-oauth@2.0.0-rc.0
# or
yarn add @fabianlars/tauri-plugin-oauth@2.0.0-rc.0
```

File: README.md [hunk 4]
```diff
@@ -1,8 +1,137 @@
...
+        // Preferebly send back only the token, or nothing at all if you can handle everything else in Rust.
+        let _ = window.emit("redirect_uri", url);
+    })
+        .map_err(|err| err.to_string())
+}
+
+#[cfg_attr(mobile, tauri::mobile_entry_point)]
+pub fn run() {
+    tauri::Builder::default()
+        .plugin(tauri_plugin_shell::init())
```
Comment (FabianLars) [inline README.md]:
```suggestion
```

File: guest-js/index.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+ *
+ * initOAuth();
+ * ```
+ */
+export const oauth: OauthPlugin = {
+  start,
+  cancel,
+  onUrl,
+  onInvalidUrl,
+};
```
Comment (FabianLars) [inline guest-js/index.ts]: hmm i think we should just export the methods directly, matching most of the official plugins.

**Comment by cuervolu** [comment]
Thank you for the suggestion. I've updated the code to export the methods directly. 

I've also updated the README and usage examples to reflect this change. Please let me know if you'd like to see any further modifications.

**Comment by FabianLars** [comment]
Only the example needs to be updated for the new exports, then i think it's good to go. Thanks again :)

**Comment by cuervolu** [comment]
Oh, I forgot! But I’ve already fixed it, and at least on my PC everything works fine. Thanks for the reminder!
